### PR TITLE
refactor: use the new flow/flownode links from the bpmn semantic in the predictions demo

### LINKED ISSUE #476: [REFACTOR] Use the new flow/flownode links from the BPMN Semantic in the "Predictions" demo
**Note**: done as part or after #260

Currently, the demo list all edges and shapes that need to be blurred. They correspond to the already executed paths.
We could only reference the Shapes and deduct the edges from the _incoming/outgoing_ data returned by the `BpmnElementsRegistry` APIs. We could also store the Edges and get the related Shapes from the _sourceRef/targetRef_.

The new information are available as of https://github.com/process-analytics/bpmn-visualization-js/releases/tag/v0.31.0 and have been used in the `draw-path` demo (see  #468).

### PR DESCRIPTION
**Pull request by csouchet**
**Live environment of examples**

https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/476-Use_the_new_flow/flownode_links_from_the_BPMN_Semantic_in_the__Predictions__demo/demo/predictions/index.html

last commit: https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/2eebd856f88103797eabea27035cc309e23f2d6a/demo/predictions/index.html

Closes #476

### CODE REVIEW

**Comment by tbouffard** [comment]
I cannot review this PR until #514 is merged. There are commits coming from the dependent PRs and new commits in this one.
This make the review hard to do. So I will wait for the merge of the other PRs.

File: demo/predictions/js/path.js
Comment (tbouffard) [inline demo/predictions/js/path.js:2]: **issue:** we cannot duplicate the code from the demo.
We already know it requires refactoring at least for https://github.com/process-analytics/bonita-day-demo-2023/issues/52. There are also a lot of extra validation of the incoming/outgoing (verification that a related edge id exists in the model) whereas this should be guarantee by bpmn-visualization.
In addition, we plan to extract PathResolver from the bonita day demo https://github.com/process-analytics/bonita-day-demo-2023/issues/36. If we duplicate the implementation here, we must track it and plan to use the shared code here. And this may not possible out of the box depending on the way we package the shared code. In particular, I don't plan to provide an IIFE bundle for the shared lib.

**Review by tbouffard — CHANGES_REQUESTED** [review]
Let's talk about alternatives to the code duplicated from the bonita day demo (PathResolver).
As I mentioned in https://github.com/process-analytics/bpmn-visualization-examples/pull/514#pullrequestreview-1494607719, it's becoming increasingly clear to me that we should consider moving this demo to its own repository to benefit from the package manager, the shared library and the framework.

File: demo/predictions/js/path.js
Comment (csouchet) [reply demo/predictions/js/path.js:2]: To replace in another PR. See #260

**Review by tbouffard — APPROVED** [review]
After discussions with @csouchet, we agreed to merge this PR as it is today.
#260 includes a task to use the shared code of `PathResolver` when available.
We also agreed to review the extraction of this demo in a dedicated repository later.

**Pull request merged**
